### ISSUE TITLE
Introduce Icons 

### ISSUE THREAD

**Issue by LHinson** [patternfly/patternfly-next#137]
- [x] Need to evaluate current PF3 and determine what icons we want to introduce. 
- [ ] Look a the class for ensuring that icons have a uniform height and width. 
- [ ] This story requires someone familiar with JS to determine how we will deliver custom patternfly icons. 

Notes:
- We already have the mechanism in place for loading FontAwesome. Just need to ensure that everything is in place to do so. 
- Visual evaluation and technical approach can happen simultaneously. 
- Add issues for anything that will be more complex from a visual evaluation

Assignees : 
Visuals: @kybaker 
CSS/JS: @mindreeper2420

**Comment by kybaker** [comment]
This issue is currently blocked by legal. I will update when we have a resolution.

**Comment by mindreeper2420** [comment]
We can set the system to use Font Awesome icons in multiple ways.

1) As an embedded stylesheet (similar to PatternFly 3), where Font Awesome is distributed with PatternFly-Next.
2) As a CDN, where the user will reference an external file, hosted by Font Awesome, in order to use the icons.
3) Allow the user to exclude any usage of Font Awesome. This may break icons, but that would be up to the user to determine (some sites/applications may not want the extra size that Font Awesome adds to an environment).

For icons created by the PatternFly Design Team, those would be distributed with PatternFly-Next and included by default.

**Comment by mindreeper2420** [comment]
I've put together a Sketch document (posted to InVIsion) comparing the current icons used in PatternFly 3 (both Font Awesome 4 and custom PatternFly icons) against what is available in Font Awesome 5 (free).

Of the 132 currently used in PatternFly, 49 do not have a viable option in Font Awesome 5. As such, those will need to be ported over to PatternFly-Next as SVGs, in order to fit with the new Font Awesome methodology (all icons are converted to SVGs upon site render).

InVision link to icon sheet: https://redhat.invisionapp.com/project/14449136

**Comment by kybaker** [comment]
I have refined Adams document to separate which icons will be transitioning and which will remain the same. A majority of the custom PatternFly icons will remain the same for now but should transition to the bolder styling in time.

https://redhat.invisionapp.com/share/AYK7M74H3CR#/301078429_PatternFly_Icons

**Comment by mindreeper2420** [comment]
I've updated my branch to include only the Font Awesome 5 icons that are listed in the doc. The users will be either just take the icons that are necessary for PatternFly-Next, or be able to update the configuration to use all Font Awesome 5 icons.

**Comment by mindreeper2420** [comment]
This issue will introduce all of Font Awesome 5 (free) to PatternFly-Next, with a followup story to handle pulling out the icons that we do not need.

**Comment by dgutride** [comment]
Closing via #305 - follow on story in #304
